pkg/cache/cache.go

```diff
@@ -260,19 +259,27 @@ func (c *Cache) DeleteResourceFlavor(rf *kueue.ResourceFlavor) sets.Set[string]
 	return c.updateClusterQueues()
 }
 
-func (c *Cache) AddOrUpdateTopologyForFlavor(topology *kueuealpha.Topology, flv *kueue.ResourceFlavor) sets.Set[string] {
+func (c *Cache) AddOrUpdateTopologyForFlavor(flv *kueue.ResourceFlavor) bool {
```

Comment:
I feel uneasy about changing the cache for the needs of this PR.

Reply:
The idea of this PR is to optimize topology and move it to cache.

Reply:
Done